astropy/io/fits/hdu/compressed/compressed.py

```diff
@@ -1038,23 +741,58 @@ def section(self):
         """
         return CompImageSection(self)
 
-    @property
-    def tile_shape(self):
-        """
-        The tile shape used for the tiled compression.
+    # FIXME: determine how we can simplify or avoid the following methods. These
+    # seem to be needed for the checksum calculations to work properly.
 
-        This shape is given in Numpy/C order
-        """
-        return tuple(
-            [
-                self._header[f"ZTILE{idx + 1}"]
-                for idx in range(self._header["ZNAXIS"] - 1, -1, -1)
-            ]
+    def _calculate_datasum(self):
+        return self._bintable._calculate_datasum()
+
+    def _calculate_checksum(self, datasum, checksum_keyword="CHECKSUM"):
+        return self._bintable._calculate_checksum(
+            datasum, checksum_keyword=checksum_keyword
         )
 
+    def _verify_checksum_datasum(self):
+        return self._bintable._verify_checksum_datasum()
+
+    def verify_checksum(self):
+        return self._bintable.verify_checksum()
+
+    def verify_datasum(self):
+        return self._bintable.verify_datasum()
+
+    def _verify(self, *args, **kwargs):
+        if self._bintable is None:
+            return super()._verify(*args, **kwargs)
+        else:
+            return self._bintable._verify(*args, **kwargs)
+
     @property
-    def compression_type(self):
-        """
-        The name of the compression algorithm.
-        """
-        return self._header.get("ZCMPTYPE", DEFAULT_COMPRESSION_TYPE)
+    def _data_offset(self):
+        if self._bintable is not None:
+            return self._bintable._data_offset
+
+    @_data_offset.setter
+    def _data_offset(self, value):
+        if self._bintable is not None:
```

Comment:
Yes good point this is used when reading from a file

Reply:
Fixed

Reply:
So this does appear to be needed, because during initialization in the ``_BaseHDU`` class we do:

```
        self._header_offset = None
        self._data_offset = None
        self._data_size = None
```

and this gets run regardless of whether ``_bintable`` is defined or not. So I've left this as-is for now.